webpack.config.js

```diff
@@ -109,68 +151,73 @@ var options = {
                 description: process.env.npm_package_description,
                 version: process.env.npm_package_version,
                 ...JSON.parse(content.toString()),
-                content_security_policy: content_security_policy,
               })
             );
           },
         },
       ],
-      {
-        logLevel: 'info',
-        copyUnmodified: true,
-      }
-    ),
-    new CopyWebpackPlugin(
-      [
+    }),
+    new CopyWebpackPlugin({
+      patterns: [
         {
-          from: 'src/pages/Content/content.styles.css',
+          from: 'src/pages/ContentScripts/content.styles.css',
           to: path.join(__dirname, 'build'),
           force: true,
         },
       ],
-      {
-        logLevel: 'info',
-        copyUnmodified: true,
-      }
-    ),
-    new CopyWebpackPlugin(
-      [
+    }),
+    new CopyWebpackPlugin({
+      patterns: [
         {
-          from: 'src/assets/img/',
+          from: 'src/assets/img/main.png',
           to: path.join(__dirname, 'build'),
           force: true,
         },
       ],
-      {
-        logLevel: 'info',
-        copyUnmodified: true,
-      }
-    ),
+    }),
     new HtmlWebpackPlugin({
       template: path.join(__dirname, 'src', 'pages', 'Options', 'index.html'),
       filename: 'options.html',
       chunks: ['options'],
+      cache: false,
     }),
     new HtmlWebpackPlugin({
       template: path.join(__dirname, 'src', 'pages', 'Popup', 'index.html'),
       filename: 'popup.html',
       chunks: ['popup'],
+      cache: false,
     }),
-    new WriteFilePlugin(),
-    new CrxWebpackPlugin({
-      keyFile: 'build.pem',
-      contentPath: 'build',
-      outputPath: 'release',
-      name: 'hypercrx',
+    // Work around for Buffer is undefined:
+    new webpack.ProvidePlugin({
+      Buffer: ['buffer', 'Buffer'],
     }),
   ],
+  infrastructureLogging: {
+    level: 'info',
+  },
+
   node: {
-    fs: 'empty',
+    global: true,
   },
 };
 
-if (ENV.NODE_ENV === 'development') {
-  options.devtool = 'cheap-module-eval-source-map';
+if (env.NODE_ENV === 'development') {
+  options.devtool = 'cheap-module-source-map';
+} else {
+  options.optimization = {
+    minimize: true,
+    minimizer: [
+      new TerserPlugin({
+        extractComments: false,
+        terserOptions: {
+          // avoid all constructor.name to be 't' after code compression in production mode,
+          // because the same constructor.name will lead to only one key in Features(a Map() obj),
+          // which then faild the inject2Perceptor step.
+          keep_fnames: true,
+        },
+      }),
+    ],
+  };
```

Comment:
"keep_fnames: ture" option is for a really-hard-to-detect problem:
<img width="878" alt="image" src="https://user-images.githubusercontent.com/32434520/154920469-1d62ddd2-a5b4-44e1-84d3-d6da6237e34e.png">

As we know, webpack will try its best to minimize the bundles size in build mode and one of its weapons is to substitute long variable names with short ones. Each module share a common name transformation rule because each module has its own name space when impoted properly. 

The problem in our code is, we import files directly by `import './xxx'`, this makes 6 transformed _t_ to be exposed in same name space and then they conflict.

So keep founction names when building bundles.